src/Education/Commands/RuleDescriptionToolWindow.cs

```diff
@@ -20,20 +20,34 @@
 
 using System;
 using System.Runtime.InteropServices;
+using System.Windows.Controls;
+using System.Windows.Documents;
 using Microsoft.VisualStudio.Shell;
 
 namespace SonarLint.VisualStudio.Education.Commands
 {
+    public interface IRuleDescriptionToolWindow
+    {
+        void UpdateContent(FlowDocument newContent);
+    }
+
     [Guid(ToolWindowIdAsString)]
     internal class RuleDescriptionToolWindow : ToolWindowPane
     {
         private const string ToolWindowIdAsString = "9E74B368-9FC3-47B0-A1C7-2DBA3A2C1762";
         public static readonly Guid ToolWindowId = new Guid(ToolWindowIdAsString);
+        private readonly FlowDocumentReader flowDocumentReader;
 
         public RuleDescriptionToolWindow()
         {
             Caption = Resources.RuleDescriptionToolWindowCaption;
-            Content = new RuleDescriptionUserControl();
+            flowDocumentReader = new FlowDocumentReader();
+            Content = flowDocumentReader;
```

Comment:
I'm not sure if there are any threading issues we need to be aware of here i.e. do we need to make sure the tool window is created/updated on the UI thread?

However, it's going to be hard to check until we're actually populating the window for real in the IDE, so I think we can leave for now (although it would be worth adding a couple of `TODO: check if UI thread requried` here and in the `UpdateContent` method).

Reply:
Ah - the thread handling is in the education service, which makes sense.